packages/solidity-contracts/contracts/fuelchain/FuelMessagePortal/v3/FuelMessagePortalV3.sol

```diff
@@ -79,31 +79,21 @@ contract FuelMessagePortalV3 is FuelMessagePortalV2 {
      * @notice Resets the rate limit amount.
      * @param _amount The amount to reset the limit to.
      */
-    function resetRateLimitAmount(uint256 _amount) external onlyRole(SET_RATE_LIMITER_ROLE) {
-        uint256 withdrawalLimitAmountToSet;
-        bool amountWithdrawnLoweredToLimit;
-        bool withdrawalAmountResetToZero;
-        
+    function resetRateLimitAmount(uint256 _amount) external onlyRole(SET_RATE_LIMITER_ROLE) {   
+        // currentPeriodAmount is not updated when duration hasn't passed to avoid a edge case scenario where extra amount can be withdrawn from the bridge when it shouldn't be
```

Comment:
Instead of having these long comments, please add a new section in `ARCHITECTURE.md`:

- Let's give credits to the original authors of this (Linea bridge), put links to the original implementation. Do this in the code as well, following natspec
- In the markdown, highlight the differences between Linea and our implementation.
- Describe an scenario and the outcomes, using both Linea 's and our implementation.